streaming_providers/debrid_client.py

```diff
@@ -140,14 +141,20 @@ async def _handle_service_specific_errors(self, error_data: dict, status_code: i
 
     @staticmethod
     async def _parse_response(
-        response: ClientResponse, is_return_none: bool, is_expected_to_fail: bool
+        response: ClientResponse, is_return_none: bool, is_expected_to_fail: bool, is_http_response: bool = False
     ) -> Union[dict, list, str]:
         if is_return_none:
             return {}
         try:
+            if is_http_response:
+                response.body = await response.json()
```

Comment:
I did that initially... but since the response is used with `with`

https://github.com/mhdzumair/MediaFusion/blob/7a9e4c5076324fa1337f2bda17a1d918729b78da/streaming_providers/debrid_client.py#L68-L70

by the time we'll be calling `.json` on the returned `reponse` object, the response body won't be available to read anymore.

Reply:
Ohh I see. Then it should be good.